src/app/search/search-filter.ts

```diff
@@ -161,28 +168,116 @@ function makeSearchFilterFactory(
           const filterName = ast.type;
           const filterValue = ast.args;
 
-          // "is:" filters are slightly special cased
-          const filterDef = filterName === 'is' ? isFilters[filterValue] : kvFilters[filterName];
-
-          if (filterDef) {
-            // Each filter knows how to generate a standalone item filter function
-            try {
-              return filterDef.filter({ filterValue, ...filterContext });
-            } catch (e) {
-              // TODO: mark invalid - fill out what didn't make sense and where it was in the string
-              errorLog('search', 'Invalid query term', filterName, filterValue, e);
-              return undefined;
+          if (filterName === 'is') {
+            // "is:" filters are slightly special cased
+            const filterDef = isFilters[filterValue];
+            if (filterDef) {
+              try {
+                return filterDef.filter({ lhs: filterName, filterValue, ...filterContext });
+              } catch (e) {
+                // An `is` filter really shouldn't throw an error on filter construction...
+                errorLog(
+                  'search',
+                  'internal error: filter construction threw exception',
+                  filterName,
+                  filterValue,
+                  e
+                );
+              }
+            }
+            return undefined;
+          } else {
+            const filterDef = kvFilters[filterName];
+            const matchedFilter = matchFilter(filterDef, filterName, filterValue);
+            if (matchedFilter) {
+              try {
+                return matchedFilter(filterContext);
+              } catch (e) {
+                // If this happens, a filter declares more syntax valid than it actually accepts, which
+                // is a bug in the filter declaration.
+                errorLog(
+                  'search',
+                  'internal error: filter construction threw exception',
+                  filterName,
+                  filterValue,
+                  e
+                );
```

Comment:
One goal of this PR is that if a filter RHS matches a declared syntax of the filter, then that filter better produce a useful predicate. This could be upgraded to a `reportException` if we think there might still be bugs lurking in the filter constructors...